nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/FractionalSecondsUtils.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.serialization.record.util;
+
+import java.time.Instant;
+
+public class FractionalSecondsUtils {
+    private static final long YEAR_TEN_THOUSAND = 253_402_300_800_000L;
+
+    private FractionalSecondsUtils() {
+    }
+
+    public static Instant tryParseAsNumber(final String value) {
+        // If decimal, treat as a double and convert to seconds and nanoseconds.
+        if (value.contains(".")) {
+            final double number = Double.parseDouble(value);
+            return toInstant(number);
+        }
+
+        // attempt to parse as a long value
+        final long number = Long.parseLong(value);
```

Comment:
Promoting this parsing to a public method moves the exception handling requirements to the caller, which could be unexpected for referencing classes. I recommend catching the NumberFormatException in this class and throwing the FieldConversionException.

Reply:
Currently `FieldConversionException` is protected, should I make both constructors public?

Reply:
I would just move the utility class into the same `field` package in this case.